### DISCUSSION TITLE
[Request for Sample] Xaml-Markup + Mvux + Navigation => use x:Bind to bind xaml propertys like .Text from e.g. MainModel

### DISCUSSION THREAD

**Discussion by DevTKSS** [unoplatform/uno.extensions#2622] [Q&A]
Uno recommends to use x:Bind for Binding, but all Samples I found were using c# Markup which has set the Datacontext in one of the first lines by default or if the sample did use xaml and mvux, it did not use navigation.
As far as I understood until now (learning, thats why I look for a sample) is that, if I am using the Navigator, the link between the e.g. MainPage and MainViewModel is did by that and with Binding which is used in the Uno App Template and uses Reflection instead of the higher Performant x:Bind is able to bind with that to the MainModel.
I dont know, if I have to instance MainViewModel and MainModel additionally on that scenario, since the Weather App sample uses a Service which is not this record thing MainModel would be...

Would anyone be able to add a template for the recommended x:Bind for this setup of a uno app? Thanks in advance!

**Comment by jeromelaban** [comment]
Can you provide a sample of what you're trying to do, and where the current documentation fails?

**Reply by DevTKSS** [reply]
clicked through the Extensions\mvux documentation area:
App Examples on e.g. Working with records: SimpleCalc is using FrameNavigation, which is not adding the DataContext like Uno.Navigator:
![image](https://github.com/user-attachments/assets/96a691b3-8e65-4f21-a024-7401c35efe39)
The Example No2 which is named there is the ChatGPT Sample, which is using navigation, but that is using c# markup with mvux, so in c# the DataContext is set in the very top like you know.
=> No example for navigation + xaml on that combined with mvux

all of the main mvux folder of docs did use Binding when it came to xaml Binding to any of the explained properties.

In this nested area I did find usage of x:Bind, on the Pagination app mvux example, but there is no navigation used (also looked up the repo of that example) so it is setting "DataContext = new PeopleViewModel(new PeopleService());" while initializing the MainPage. With navigation this Datacontext should be already provided as far as I understood the above navigation setup in App.xaml.cs, correct? So there should not be typed that line in that scenario. But still 90% of the xaml code examples all around this area are using Binding instead of the x:Bind you told to prefer... I am trying, looking for the "how to" for that app, because I am trying to learn how to build app with xaml but would like to use the navigation and mvux uno enables.

so swapped to search keyword "x:Bind" which brought me to "Reference/UI/Features/x:Bind" which shows many examples for xaml with x:Bind, but the only one which might could be made for this was the one listed on Pathless Casting. Problem on that, it doesnt work as expected, althought the Properties and Methods has been provided in the publicpartial record MainModel.
![Screenshot 2024-11-18 213059](https://github.com/user-attachments/assets/3296401f-cfbc-432b-9cda-dbc93caa0fe9)
![Screenshot 2024-11-18 213751](https://github.com/user-attachments/assets/30b4aeb1-8047-4efa-a556-2ccf79647d29)
![image](https://github.com/user-attachments/assets/0f8ce350-5e15-4173-a7e5-7308610c2da8)
header of that MainPage as default:
![image](https://github.com/user-attachments/assets/64b69a53-7215-4923-8a8c-03fc1a73ff53)

So, to me, following the provided documentation, there has to be something missing that would provide the actual DataContext which should be brought with the uno navigator to link that, isnt it?
In Mvvm there would be something like that "private MainViewModel ViewModel => (MainViewModel)DataContext;" in MainPage.xaml.cs file and in the xaml to set "d:DataContext="{DesignInstance local:MainViewModel}", but that is actually not working with x:Bind in TwoWay Mode.

And that is why I would like to have a sample like you did for the ones from blanc preset, just for this case...
Microsoft Documentation is actually telling, That x:Bind is not available for accessing DataContext because its type of object.
In no sample uno is using the DesignInstance is inserted in the xaml header, so most likly also this is wrong.
For example, my not successfull tryes are uploaded in this repo: https://github.com/DevTKSS/MyUnoXamlApp.git
I would not mind if you want to take it for the sample to use x:Bind for all of the Binding Actions. On the MainPage.cs you can see in addition to that, the only chance for me at the moment (UI Thread issue persisting that was filed before) to get a ContentDialog be shown when clicking on a Button for example call.
with the x:Bind try to get that working my app not longer Builds and is telling me, that for MainPage Bindings is not known and much more issues that seems to be caused from this... 

hope it might get clearer... I hoped that anyone is able to get this navigator + mvux + xaml working with x:Bind, otherwise it seems to me, that this is not useable actually :(

**Comment by DevTKSS** [comment]
Closing this, seems like its caused from the common known x:Bind Roslyn Generators issue and its just not possible because of this to use x:Bind together with Mvux and the Navigator.
